apps/explorer/src/components/module/module-functions-interaction/useTxEffectsObjectRefs.ts

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { getTransactionEffects } from '@mysten/sui.js';
+import { useMemo } from 'react';
+
+import type { SuiTransactionResponse } from '@mysten/sui.js';
+
+export function useTxEffectsObjectRefs(
```

Comment:
I saw us doing this in other places in the explorer and tbh I'm kind of unconvinced this needs to be a memoized pattern anyway. A map is really cheap computationally and we probably don't need referential equality. You could also just update `toObjectLink` to do the reference lookup for you.